doc/design.md

```diff
@@ -0,0 +1,128 @@
+
+# Levity remote task runner  
+
+## High-Level Description
+
+`Levity` is a (small part of a) distributed task-running system, consisting of:
+
+1. a server application that can schedule and monitor the execution of arbitrary processes, & an API that exposes that functionality, and
+2. a client application that uses the API to interact with the service.
+
+## Intended Use
+
+`Levity` is intended to be used as an exercise as part of the Gravitational interview process. As this is an exercise, most design tradeoffs will be made in favour of simple implementation over producing an infinitely-robust, battle-hardened production system.
+
+That said, the code that *is* implemented is expected to be robust, clean, modular, and race-free.
+
+## Implementation Language
+
+Go 1.15
+
+## Repo Layout 
+
+The project will use the [Within Go Repo Layout](https://christine.website/blog/within-go-repo-layout-2020-09-07).
+
+## RPC mechanism
+
+`Levity` will use gRPC as its underlying RPC mechanism. The reasons for this are:
+
+* gRPC has built-in support for secure transport and authentication mechanisms, obviating the need to implement these features independently, or wrap them manually around a web service,
+* gRPC mandates the use of a Protocol Buffer description of the service, and having an implementation-independent source of truth about the API is always useful, and
+* I've never used it before, so it seems as good a time to learn as any.
+
+## API
+
+The API exposed by the server is described by the protocol buffer specification [here](../api/levity.proto).
+
+## Task Lifecycle
+
+1. Task is started by a call to `StartTask`.
+2. The client repeatedly polls server for task completion via `QueryTask`.
```

Comment:
Can you clarify? This is going to be an option for the client to do, not something the client will be doing normally right?

Basically you plan to implement `./client status uuid` that will return right away with the status but the client can be run in a loop if the caller wants with something like `watch`?

Reply:
Essentially yes. Will change the wording to indicate that the user may use the client to do this if they want to monitor a task as it executes.